frontend/src/components/Catalog.js

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from 'react';
+import {
+  Card,
+  CardImg,
+  CardText,
+  CardBody,
+  CardTitle,
+  CardSubtitle,
+  Button
+} from 'reactstrap';
+import { Container, Row, Col } from 'reactstrap';
+import PropTypes from 'prop-types';
+import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+
+class Catalog extends Component {
+  state = {
+    page: 1
+  };
+
+  getFirstRow = () => {
+    var page = this.state.page;
+    var start = (page - 1) * 6;
+    return this.props.cards.slice(start, start + 3);
+  };
+
+  getSecondRow = () => {
+    var page = this.state.page;
+    var start = (page - 1) * 6;
+    return this.props.cards.slice(start + 3, start + 6);
+  };
+
+  changePage = value => {
+    this.setState({ page: value });
+    console.log(value);
+  };
+
+  renderPaginationItems = () => {
+    var length = this.props.cards.length;
+    var numPageItems = Math.ceil(length / 6);
+    let numArr = [];
+
+    for (var i = 1; i <= numPageItems; i++) {
+      numArr.push(i);
+    }
+
+    return numArr.map(i => (
+      <PaginationItem key={i}>
+        <PaginationLink onClick={() => this.changePage(i)}>{i}</PaginationLink>
+      </PaginationItem>
+    ));
+  };
+
+  render() {
+    return (
+      <div>
+        <Container fluid={true}>
+          <Row>
+            {this.getFirstRow().map(x => (
+              <Col lg="4">
+                <Card>
+                  <CardBody>
+                    <CardTitle>{x.title}</CardTitle>
+                    <CardSubtitle>{x.subtitle}</CardSubtitle>
+                    <CardText>{x.text}</CardText>
+                  </CardBody>
+                </Card>
+              </Col>
+            ))}
+          </Row>
+          <Row>
+            {this.getSecondRow().map(x => (
+              <Col lg="4">
+                <Card>
+                  <CardBody>
+                    <CardTitle>{x.title}</CardTitle>
+                    <CardSubtitle>{x.subtitle}</CardSubtitle>
+                    <CardText>{x.text}</CardText>
+                  </CardBody>
+                </Card>
+              </Col>
+            ))}
+          </Row>
+          <Row>
+            <Pagination size="lg" aria-label="Page navigation example">
+              <PaginationItem>
+                <PaginationLink previous href="#" />
```

Comment:
These `previous` and `next` links also need an `onClick` function in order to work.

For this, we won't be able to call `this.setState` with an object like we've been doing. Instead, you need to pass it an `updater` function. You can read about it here: https://reactjs.org/docs/react-component.html#setstate

Here's a sample code of what we'd want to do:
```
onClickPrevious = () => this.setState((state) => {
  return {page: state.page == 1 ? state.page : state.page - 1};
}
```